pkg/kubelet/dockertools/docker_manager.go

```diff
@@ -716,13 +720,23 @@ func (dm *DockerManager) runContainer(
 
 // setInfraContainerNetworkConfig sets the network configuration for the infra-container. We only set network configuration for infra-container, all
 // the user containers will share the same network namespace with infra-container.
-func setInfraContainerNetworkConfig(pod *api.Pod, netMode string, opts *kubecontainer.RunContainerOptions, dockerOpts *dockertypes.ContainerCreateConfig) {
+func (dm *DockerManager) setInfraContainerNetworkConfig(pod *api.Pod, netMode string, opts *kubecontainer.RunContainerOptions, dockerOpts *dockertypes.ContainerCreateConfig) {
 	exposedPorts, portBindings := makePortsAndBindings(opts.PortMappings)
 	dockerOpts.Config.ExposedPorts = exposedPorts
 	dockerOpts.HostConfig.PortBindings = dockernat.PortMap(portBindings)
 
+	result, err := dm.checkDockerAPIVersion(dockerV112APIVersion)
+	if err != nil {
+		glog.Errorf("Failed to check docker api version: %v", err)
+	}
+
+	hostNetPlusDNSEnabled := result >= 0
+
 	if netMode != namespaceModeHost {
 		dockerOpts.Config.Hostname = opts.Hostname
+	}
+
+	if netMode != namespaceModeHost || (netMode == namespaceModeHost && hostNetPlusDNSEnabled) {
```

Comment:
> 3. --net=host + dns settings already are available thru DOCKER_OPTS for a while
> 4. so the patch added fro docker 1.12 is about to enable usage of --net=host and --dns thru cli

So, if I understand points 3 and 4 correctly, the DOCKER_OPTS were already there but this adds them via the Docker CLI/API ?